File: src/main/java/com/rpb/reservation/staffhome/application/StaffHomeOverview.java
package com.rpb.reservation.staffhome.application;

import java.time.LocalDate;
import java.util.List;
import java.util.UUID;

public record StaffHomeOverview(
    UUID storeId,
    LocalDate businessDate,
    String storeTimezone,
    ReservationMetrics reservation,
    QueueMetrics queue,
    TableMetrics tables,
    List<PartySizeGroupMetrics> partySizeGroups
) {

    public StaffHomeOverview {
        partySizeGroups = partySizeGroups == null ? List.of() : List.copyOf(partySizeGroups);
    }

    public record ReservationMetrics(
        int totalReservations,
        int totalPartySize,
        int arrivedReservations,
        int arrivedPartySize,
        int seatedReservations,
        int seatedPartySize,
        int cancelledReservations
    ) {
    }

    public record QueueMetrics(
        int waitingTickets,
        int waitingPartySize,
        int calledTickets,
        int calledPartySize,
        int seatedTickets,
        int skippedTickets,
        int cancelledTickets,
        int expiredTickets
    ) {
    }

    public record TableMetrics(
        int totalTables,
        int availableTables,
        int reservedTables,
        int occupiedTables,
        int cleaningTables,
        int temporaryGroups
    ) {
    }

    public record PartySizeGroupMetrics(
        String label,
        int groups,
        int partySize
    ) {
    }
}

File: src/types/staffHomeOverview.ts
export interface StaffHomeReservationMetrics {
  totalReservations: number
  totalPartySize: number
  arrivedReservations: number
  arrivedPartySize: number
  seatedReservations: number
  seatedPartySize: number
  cancelledReservations: number
}

export interface StaffHomeQueueMetrics {
  waitingTickets: number
  waitingPartySize: number
  calledTickets: number
  calledPartySize: number
  seatedTickets: number
  skippedTickets: number
  cancelledTickets: number
  expiredTickets: number
}

export interface StaffHomeTableMetrics {
  totalTables: number
  availableTables: number
  reservedTables: number
  occupiedTables: number
  cleaningTables: number
  temporaryGroups: number
}

export interface StaffHomePartySizeGroupMetrics {
  label: string
  groups: number
  partySize: number
}

export interface StaffHomeOverviewResponse {
  success: true
  storeId: string
  businessDate: string
  storeTimezone: string
  reservation: StaffHomeReservationMetrics
  queue: StaffHomeQueueMetrics
  tables: StaffHomeTableMetrics
  partySizeGroups: StaffHomePartySizeGroupMetrics[]
}

export interface StaffHomeOverviewQuery {
  businessDate?: string
}

export interface StaffHomeOverviewApiErrorBody {
  code: string
  messageKey: string
  details: Record<string, unknown>
}

export interface StaffHomeOverviewApiErrorResponse {
  success: false
  error: StaffHomeOverviewApiErrorBody
}

export type StaffHomeOverviewApiResponse =
  | StaffHomeOverviewResponse
  | StaffHomeOverviewApiErrorResponse
